src/watcherEx.ts

```diff
@@ -0,0 +1,230 @@
+// Copyright 2022 The Casbin Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { RedisClusterConnection, RedisConnection, RedisClient } from './redis';
+import { ClusterNode, ClusterOptions, RedisOptions } from 'ioredis';
+import { Model, WatcherEx } from 'casbin';
+
+export interface WatcherExOptions extends RedisOptions {
+  channel?: string;
+}
+
+export interface WatcherExClusterOptions extends ClusterOptions {
+  channel?: string;
+}
+
+export interface WatcherExCallBackOptions {
+  addPolicyCb?: (args: {sec: string, ptype: string, params: string[]}) => Promise<void>;
+  removePolicyCb?: (args: {sec: string, ptype: string, params: string[]}) => Promise<void>;
+  removeFilteredPolicyCb?: (args: {sec: string, ptype: string, field: number, fieldValues: string[]}) => Promise<void>;
+  savePolicyCb?: (args: {policies: string[][], gPolciies: string[][]}) => Promise<any> | any;
```

Comment:
Why is this type `Promise<any> | any`, not `Promise<void>`?

Reply:
I'm sorry I must have missed it, rn I'm trying a way to fix tests